File: next-tauri/src/platform/contracts/nativeDesktop.ts
import type { ImageAssetSummary } from './image'

export interface ScreenCaptureResult {
  assets: ImageAssetSummary[]
  monitorCount: number
}

export interface ScreenColorSample {
  hex: string
  red: number
  green: number
  blue: number
  x: number
  y: number
}

export interface DisplaySleepStatus {
  active: boolean
  owned: boolean
}

export interface NativeDesktopApi {
  captureDisplays(delayMs?: number): Promise<ScreenCaptureResult>
  sampleScreenColor(delayMs?: number): Promise<ScreenColorSample>
  getDisplaySleepStatus(owner: string): Promise<DisplaySleepStatus>
  setDisplaySleepPrevention(owner: string, enabled: boolean): Promise<DisplaySleepStatus>
}

File: next-tauri/src-tauri/src/contracts/native_desktop.rs
use serde::{Deserialize, Serialize};

use super::image::ImageAssetSummary;

#[derive(Clone, Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ScreenCaptureResult {
    pub assets: Vec<ImageAssetSummary>,
    pub monitor_count: usize,
}

#[derive(Clone, Debug, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ScreenColorSample {
    pub hex: String,
    pub red: u8,
    pub green: u8,
    pub blue: u8,
    pub x: i32,
    pub y: i32,
}

#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DisplaySleepStatus {
    pub active: bool,
    pub owned: bool,
}
